src/orion/algo/tpe.py

```diff
@@ -185,12 +185,13 @@ def __init__(self, space, seed=None,
                                   full_weight_num=full_weight_num)
 
         for dimension in self.space.values():
+
             if dimension.type not in ['real', 'integer', 'categorical']:
                 raise ValueError("TPE now only supports Real, Integer "
                                  "and Categorical Dimension.")
 
-            if dimension.prior_name not in ['uniform', 'int_uniform', 'choices']:
-                raise ValueError("TPE now only supports uniform, uniform discrete "
+            if dimension.prior_name not in ['uniform', 'reciprocal', 'int_uniform', 'choices']:
```

Comment:
We could support int_reciprocal as well.

Reply:
I saw this possiblity, but did not want to do this now for 2 reasons: 1)discrete int solution still need verify, 2) loguniform is a little bit tricky for the space range, and user needs be careful to specify the space, didn't want to make it more complicated, plus I think it is almost good enough to have uniform support for discrete it.

Reply:
Agreed!